amazon/ion/writer_binary_raw.py

```diff
@@ -0,0 +1,365 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at:
+#
+#    http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Writer for raw binary Ion values, without symbol table management."""
+
+# Python 2/3 compatibility
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import struct
+from decimal import Decimal
+from datetime import datetime
+from functools import partial
+
+import six
+
+from .writer_binary_raw_fields import _write_varuint, _write_uint, _write_varint, _write_int
+from .core import IonEventType, IonType, DataEvent, Transition
+from .util import coroutine, Enum
+from .writer import writer_trampoline, partial_transition, WriteEventType
+
+
+class _TypeIds(Enum):
+    null = 0x00
+    bool_false = 0x10
+    bool_true = 0x11
+    pos_int = 0x20
+    neg_int = 0x30
+    float = 0x40
+    decimal = 0x50
+    timestamp = 0x60
+    symbol = 0x70
+    string = 0x80
+    clob = 0x90
+    blob = 0xA0
+    list = 0xB0
+    sexp = 0xC0
+    struct = 0xD0
+    annotation_wrapper = 0xE0
+
+
+class _Zeros(Enum):
+    """Single-octet encodings, represented by the type ID and a length nibble of zero.
+
+    Notes:
+        Blob, clob, list, and sexp also have single-octet encodings, but the current implementation handles these
+        implicitly.
+    """
+    int = _TypeIds.pos_int | 0x0  # Int zero is always encoded using the positive type ID.
+    float = _TypeIds.float | 0x0  # Represents zero.
+    decimal = _TypeIds.decimal | 0x0  # Represents 0d0.
+    string = _TypeIds.string | 0x0  # Represents the zero-length (empty) string.
+    symbol = _TypeIds.symbol | 0x0  # Represents symbol zero.
+    struct = _TypeIds.struct | 0x0  # Represents a struct with zero fields.
```

Comment:
Yes it is. I did this to be explicit because you had previously made a comment that it was confusing (though correct) to just write the type ID for the single-byte representation.

Is there some other way you'd prefer I address that confusion?